client/src/components/payment/payment-modal.vue

```diff
@@ -39,33 +44,60 @@ import * as paymentActions from './payment-actions'
 import PaymentMixin from '@/components/payment/payment-mixin'
 import PaymentSessionMixin from '@/components/payment/payment-session-mixin'
 import NameRequestMixin from '@/components/mixins/name-request-mixin'
+import DisplayedComponentMixin from '@/components/mixins/displayed-component-mixin'
 
 import { getBaseUrl } from './payment-utils'
 
+export const PAYMENT_COMPLETION_TIMER_NAME = 'paymentCompletionTimer'
+export const PAYMENT_COMPLETION_TIMEOUT_MS = 2 * (60 * 1000) // Set to 2 minutes
+
 @Component({
   components: {
     RequestDetails,
-    FeeSummary
+    FeeSummary,
+    CountdownTimer
+  },
+  props: {
+    onActivate: {
+      type: Function,
+      default: async () => undefined
```

Comment:
Why implement props this way instead of with `@Prop` decorator+method?

Reply:
@Prop decorator doesn't appear to cover all use cases that you can do with standard Vue syntax. Also that library used to add those @Prop are a 3rd party library not core Vue functionality IIRC.

Reply:
I'm curious about that but leave it for now.

BTW, you already import that library to provide `@Prop`... it's "vue-property-decorator'".